src/services/cognito.js

```diff
@@ -201,6 +203,7 @@ function getUserProfile() /* : {
     user.isSAMLUser = user.providerType === 'SAML'
     if (user.isSAMLUser) {
       user.username = user.providerUserId
```

Comment:
Can we instead just make this:

```suggestion
      user.username = userProfile.preferred_username || user.providerUserId
```

We have logic else where that will display the `username` property. I think it's appropriate to just ignore the `providerUserId` if they have a preferred username setup.

Reply:
I thought about this but could not say for sure doing this wont break anything in existing submission events. It seems unlikely though, whats your thoughts?

It would also introduces a mismatch between whats in the JWT and what is stored in s3 (which is not a huge deal its just annoying that in one part of a code base `user.preferred_username` would be used however in submission events its `user.username`) - i mapped the user name claim to `preferred_username` because username must be unique in cognito, and its entirely possible that more than one customer has a user `john.smith` but there can be many `preferred_username`s with the same value

Reply:
I am 99.9% sure it wont break anything.

In regards to the username being different between our user object and the JWT, that is already the case. The `username` property is set to the user's email address for a normal username/password login. This `username` property on our object is suppose to be `The username used to login`.

Reply:
sweet as... will update the pr